### PR TITLE
Add integration tests and publishing to orb registry

### PR DESCRIPTION
**Pull request by florisvdg**
Idea is: first publish dev orb, then trigger a second pipeline which imports the newly published dev orb and uses it to run tests. And if that succeeds on a tag, publish it to the registry.

~One thing to note: the CircleCI CLI (and therefore `orb-tools` orb too) requires explicitly setting whether the release is `major`, `minor`, or `patch`. So we'll have to include that in the tag name unfortunately: e.g. `minor-v0.1.0`.~

### CODE REVIEW

**Comment by SimonBarendse** [comment]
> Idea is: first publish dev orb, then trigger a second pipeline which imports the newly published dev orb and uses it to run tests. And if that succeeds on a tag, publish it to the registry.

Can we do all of this in one pipeline if we use the SHA version of the orb in the integration tests?
The pipeline can look as follows:
publish SHA to dev => run integration tests => if on master: publish to dev:latest, if on tag publish to the new version

**Comment by florisvdg** [comment]
Orb imports happen at the beginning of a workflow, so that's why it's like this.

File: .circleci/config.yml [hunk 1]
```diff
@@ -84,14 +84,14 @@ workflows:
           orb-name: secrethub/cli
           context: publish-orb-prd
           add-pr-comment: false
-          major-release-tag-regex: '^major-v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$'
-          minor-release-tag-regex: '^minor-v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$'
-          patch-release-tag-regex: '^patch-v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$'
+          major-release-tag-regex: '^v[1-9][0-9]*\.0\.0+$'
+          minor-release-tag-regex: '^v[0-9]*\.[1-9][0-9]*\.0+$'
+          patch-release-tag-regex: '^v[0-9]*\.[0-9]*\.[1-9][0-9]*$'
```
Comment (florisvdg) [inline .circleci/config.yml:89]: Regexes are now smarter, so we can tag it the way we always do, without a prefix.

Comment (florisvdg) [reply .circleci/config.yml:89]: I tested this locally with this code: 

https://github.com/CircleCI-Public/orb-tools-orb/blob/0a48b68d679e4e720adf815935791a3ca6b20621/src/commands/dev-promote-from-git-tag.yml#L133-L143

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,9 +2,48 @@ version: 2.1
...
+      DOCKER_USERNAME: secrethub://company/app/docker/username
+      DOCKER_PASSWORD: secrethub://company/app/docker/password
+    steps:
+      - checkout
+      - secrethub/exec:
+          step-name: Validate if secrets have been provisioned
+          command: |
+            set -e
+            : ${DOCKER_USERNAME:?DOCKER_USERNAME is not set}
+            : ${DOCKER_PASSWORD:?DOCKER_PASSWORD is not set}
```
Comment (SimonBarendse) [inline .circleci/config.yml]: Do we also want to check whether they're set to the correct value? Similar to [how we test our GitHub Action](https://github.com/secrethub/actions/blob/v0.1.0/.github/workflows/main.yml#L14)

**Review by SimonBarendse — APPROVED** [review]
Way better with the "normal" vx.y.z tags :smile: 

I have one more suggestion, other than that let's go :rocket:

**Pull request merged**